builder/amazon/common/helper_funcs.go

```diff
@@ -47,3 +49,14 @@ func DestroyAMIs(imageids []*string, ec2conn *ec2.EC2) error {
 	}
 	return nil
 }
+
+// Returns true if the error matches all these conditions:
+//  * err is of type awserr.Error
+//  * Error.Code() matches code
+//  * Error.Message() contains message
+func isAWSErr(err error, code string, message string) bool {
```

Comment:
oh, this is very useful. I can think of several places in the code that could use this func.

Reply:
nice !

Reply:
This helper was pretty useful in the AWS Terraform provider so I thought we should borrow it :smile: I'll create a separate PR for this helper and update the `err.(awserr.Error)` checks to use this helper instead. There are not many references in the Amazon builder so it should be pretty straight forward.

Reply:
Separate PR for helper function https://github.com/hashicorp/packer/pull/8365